### ISSUE TITLE
onChange behaviour

### ISSUE THREAD

**Issue by ivanfr90** [wojtekmaj/react-time-picker#12]
Hi!

Reading the documentation the `onChange` event is fired only when a valid time is inserted. This could be correct, but when I delete only the minutes, the `onChange` event is not fired (because time is not valid), keeping in this conjuncture the last valid value saved.

For example:
First, I write 02:45, time is valid so `onChange` is fired --> `state.time` = '02:45'
![image](https://user-images.githubusercontent.com/4324408/45308244-51757300-b521-11e8-921b-14cd997c09a0.png)

Later, I delete only minutes, 02:--, time is invalid so `onChange` is not fired --> `state.time` = '02:45'. In this case, current time is not correct so `state.time` has the last valid time, but the input is wrong.
![image](https://user-images.githubusercontent.com/4324408/45308274-6225e900-b521-11e8-936b-e5dcc6de6b8c.png)

How it could be detected when a invalid time is inserted?

Thank you!

**Comment by wojtekmaj** [comment]
Observing the behaviour of the native time picker implemented in some modern browsers, partially clearing an input so that it's invalid (like you did) should fire onChange event with empty value. I'll consider this a bug.

**Comment by wojtekmaj** [comment]
Okay, I've done some research and it seems like default onChange behaviour just doesn't work with React very much. onChange sending empty values when it's only partially empty causes even the native time picker to clear if it's a controlled input
Given that, I'll implement support for attaching event listeners to all inputs (you can do that now but putting the picker in a wrapper which has these listeners, but that's not too obvious behaviour). On an event like keyup you'll be able to verify on your own whether the inputs are empty or not.

**Comment by ivanfr90** [comment]
I see that you are looking for a similar experience to native input time. I will try to give you my perspective :

`onChange` event should be fired when:
*  Time is correct. When will time be correct? Always if a complete time is setted: hours field, minutes field, and (if has seconds) seconds fields are filled (*). The value will be a valid time.
*  Any field is deleted. An onChange event will be fired without value. The value will be empty.

(*) Could an user input a wrong or non-existent time? Definitely, no.

The explanation it's that native input type time never allows user to set incorrect hours or minutes.
The process works as follows:
1. We have an empty input:
![image](https://user-images.githubusercontent.com/4324408/45772772-e62a5000-bc48-11e8-9342-2f202a9719fe.png)
2. User puts focus on hours and set one number, for example 1:
![image](https://user-images.githubusercontent.com/4324408/45773771-82eded00-bc4b-11e8-89e3-c33d3d7611f6.png)    
Automatically the native behavior is to skip to minutes (personally I don't like this, because if I want two set an hour of two digits, always I have to return to hours field and put focus again on it).
3. Following with the hour field. 
    * If user sets another valid digit the hour field is completed and lost focus, skipping to minutes:
    ![image](https://user-images.githubusercontent.com/4324408/45773824-a6189c80-bc4b-11e8-8a63-c6bcf86489e0.png)
    * If user sets an invalid digit the hour fields keep the previous valid digit and skip to minutes. For example I try to add the digit 3:
    ![image](https://user-images.githubusercontent.com/4324408/45773884-d19b8700-bc4b-11e8-8380-0385702a70dd.png)


In summary the behavior is'ts easy: native input time start adding digits from right to left. If an attempt to set an invalid number is detected, automatically the digit is rejected.

**Comment by chinnasamya-tudip** [comment]
Hi @wojtekmaj,
I used this to handle the invalid date.
`
class TimePicker extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
    	value: '13:58',
    }
  }
  
  onChange(event) {
    if (event.target.value.length) {
  	  this.setState({ value: event.target.value })
    }
  }
  
  render() {
    return (
      <div>
        <p>Controlled input</p>
        <input
          type="time"
          onChange={this.onChange.bind(this)}
          value={this.state.value}
         />
        <p>Uncontrolled input</p>
        <input
          type="time"
          defaultValue={this.state.value}
         />
      </div>
    )
  }
}

ReactDOM.render(<TimePicker />, document.querySelector("#app"))
`
e.target.value will not change when entering the backspace. Might be it will helpfull for you.

**Comment by reverie** [comment]
@wojtekmaj Here's a proposal. Add a prop `fullOnChange` that takes a callback with a list of 2-tuples, where the first element is time format of the kind passed to `format`—e.g. 'h', 'hh', 's', 'a—and the second element is the new value for that field, possible null. Also accept a new prop `fullValue` where I can pass in such an array. (The `value` and `fullValue` props would be mutually exclusive.) This way I could wrap `TimePicker` and pass the reconstructed time I want up to my parent component, such as `null` if any fields are null, without ruining the state of the time picker.

**Comment by agnieszka-puzio** [comment]
@wojtekmaj any updates on this? Are those listeners implemented?
